src/pcre2_compile.c

```diff
@@ -10528,7 +10536,9 @@ if ((options & PCRE2_LITERAL) == 0)
           break;
 
           default:
-          PCRE2_UNREACHABLE();
+          /* Fail assertion if a new type was added and someone forgot to
+          update this switch */
+          PCRE2_DEBUG_UNREACHABLE();
```

Comment:
Funny enough this one was one of the few new asserts that were commented, an after reading the comment think it should had been kept as `PCRE2_UNREACHABLE()`

1. In debug mode will trigger and assert, and assuming the unnamed developer that made the mistake tests locally with a debugging build then it should never reach a release. Note that we do debug builds for all the DEV CI jobs that even if them were careless enough it might be caught there.
2. While the potential for performance improvement here is irrelevant, the whole reason why this macro is even enabled in production is to optimize cases like this, so why it should have been done IMHO.

I am curious why this was changed?, and if (considering we should be well aware on how to use this macros) maybe what we are missing is documentation in HACKING (or somewhere else) that would clarify how to use them so the developer that eventually hits this should had known better?, maybe we can reuse some of the information from Alex's commit that was not used?

Reply:
FWIW added some documentation on the code (like we do for other macros/functions) so we have at least a starting point

Reply:
I see that Carlo has added a `MAX_LIST` variable in GenerateUcd.py, though it doesn't seem to be used yet in that script.

Would adding a check around line 851 that throws an exception if `len(s) > (MAX_LIST - 3)` be a good idea? Then anyone who is regenerating the property tables will notice if this is a problem.

The reason for comparing `len(s)` to `MAX_LIST - 3` is because two slots in `list` are used for other purposes, and it needs to be terminated with the special value `NOTACHAR`, so the number of slots available for codepoints are `MAX_LIST - 3`.

If this ever happens, simply increasing the value of `MAX_LIST` in pcre2_auto_possess.c and GenerateUcd.py is all that is needed... then the stack-allocated arrays used to store those values will get bigger and everything will be OK.

Reply:
Ah... sorry... these comments apply below, not here. Let me copy them to the part which they were intended to address.